contracts/src/TeleporterTokenDestination.sol

```diff
@@ -69,9 +109,25 @@ abstract contract TeleporterTokenDestination is
         require(
             feeTokenAddress_ != address(0), "TeleporterTokenDestination: zero fee token address"
         );
+        require(decimalsShift <= 18, "NativeTokenDestination: invalid decimalsShift");
         sourceBlockchainID = sourceBlockchainID_;
         tokenSourceAddress = tokenSourceAddress_;
         feeTokenAddress = feeTokenAddress_;
+        tokenMultiplier = 10 ** decimalsShift;
+        multiplyOnReceive = multiplyOnReceive_;
+    }
+
+    /**
+     * @dev Scales `value` based on `tokenMultiplier` and the direction of the transfer.
+     * Should be used for all tokens being transferred to/from other subnets.
+     */
+    function scaleTokens(uint256 value, bool isReceive) public view returns (uint256) {
```

Comment:
Was the scaling logic moved from `NativeTokenDestination` to `TeleporterTokenDestination` to directly support send and call? Or was this an unrelated refactor?

Reply:
It was unrelated to `sendAndCall`. We just thought this was a slightly cleaner approach than making it a virtual method that `ERC20Destination` and `NativeTokenDestination` provided their own implementations of.

Reply:
Cool, agreed